File: readability/readability.c
#include<cs50.h>
#include<ctype.h>
#include<math.h>
#include<stdio.h>
#include<string.h>

int count_letters(string s);
int count_words(string s);
int count_sentences(string s);
float compute_readability(float l,float s);


int main(void)
{
    string text = get_string("Text:");
    int count_l = count_letters(text);
    int count_w = count_words(text);
    int count_s = count_sentences(text);
    float average_l = ((float)count_l / count_w) * 100;
    float average_s = ((float)count_s / count_w) * 100;
    float index = compute_readability(average_l,average_s);
    //printf("Letters: %i\nWords: %i\nSentences: %i\nReadability: %.3f\n",count_l,count_w,count_s,index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if(index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n",(int)round(index));
    }

}

int count_letters(string s)
{
    int letters_count = 0;
    for (int i = 0,len = strlen(s);i < len; i ++)
    {
        if (isalpha(s[i]))
        {
            letters_count += 1;
        }
    }
    return letters_count;
}
int count_words(string s)
{
    int word_count = 1;
    for (int i = 0,len = strlen(s);i < len; i++)
    {
        if (isblank(s[i]))
        {
            word_count += 1;
        }
    }
    return word_count;
}
int count_sentences(string s)
{
    int sentence_count = 0;
    for(int i = 0,len = strlen(s);i < len; i++)
    {
        if(s[i] == '.')
        {
            sentence_count += 1;
        }
    }
    return sentence_count;
}
float compute_readability(float l,float s)
{
    float index = (0.0588 * l) - (0.296 * s) - 15.8;
    return index;
}

File: sentimental-readability/readability.py

def main():
    text = input("Text:").lower()
    count_l = count_letters(text)
    #print(count_l)
    count_w = count_words(text)
    #print(count_w)
    count_s = count_sentences(text)
    #print(count_s)
    average_l = (count_l /count_w ) * 100
    average_s = (count_s /count_w) * 100
    #print(average_l,average_s)
    index = compute_readability(average_l , average_s)
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print(f"Grade {round(index)}")

def count_letters(s):
    letters_count = 0
    for c in s:
        if c.isalpha():
            letters_count += 1
    return letters_count

def count_words(s):
    word_count = 1
    for c in s:
        if " " in c:
            word_count +=1
    return word_count

def count_sentences(s):
    sentence_count = 0
    for c in s:
        if "." in c or "!" in c or "?" in c:
            sentence_count += 1
    return sentence_count


def compute_readability(l,s):
    index = 0.0588 * l - 0.296 * s - 15.8
    return index


if __name__ == "__main__":
    main()
